source/gameengine/Ketsji/KX_TextureProbe.cpp

```diff
@@ -0,0 +1,250 @@
+{
+	return m_projection;
+}
+
+bool KX_TextureProbe::GetEnabled() const
+{
+	return m_enabled;
+}
+
+int KX_TextureProbe::GetIgnoreLayers() const
+{
+	return m_ignoreLayers;
+}
+
+float KX_TextureProbe::GetClipStart() const
+{
+	return m_clipStart;
+}
+
+float KX_TextureProbe::GetClipEnd() const
+{
+	return m_clipEnd;
+}
+
+void KX_TextureProbe::SetClipStart(float start)
+{
+	m_clipStart = start;
+}
+
+void KX_TextureProbe::SetClipEnd(float end)
+{
+	m_clipEnd = end;
+}
+
+float KX_TextureProbe::GetLodDistanceFactor() const
+{
+	return m_lodDistanceFactor;
+}
+
+void KX_TextureProbe::SetLodDistanceFactor(float lodfactor)
+{
+	m_lodDistanceFactor = lodfactor;
+}
+
+bool KX_TextureProbe::NeedUpdate()
+{
+	bool result = m_autoUpdate || m_forceUpdate;
+	m_forceUpdate = false;
+
+	return result;
+}
+
+#ifdef WITH_PYTHON
+
+PyTypeObject KX_TextureProbe::Type = {
+	PyVarObject_HEAD_INIT(NULL, 0)
+	"KX_TextureProbe",
+	sizeof(PyObjectPlus_Proxy),
+	0,
+	py_base_dealloc,
+	0,
+	0,
+	0,
+	0,
+	py_base_repr,
+	0, 0, 0, 0, 0, 0, 0, 0, 0,
+	Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
+	0, 0, 0, 0, 0, 0, 0,
+	Methods,
+	0,
+	0,
+	&CValue::Type,
+	0, 0, 0, 0, 0, 0,
+	py_base_new
+};
+
+PyMethodDef KX_TextureProbe::Methods[] = {
+	KX_PYMETHODTABLE_NOARGS(KX_TextureProbe, update),
+	{NULL, NULL} // Sentinel
+};
+
+PyAttributeDef KX_TextureProbe::Attributes[] = {
+	KX_PYATTRIBUTE_RW_FUNCTION("viewpointObject", KX_TextureProbe, pyattr_get_viewpoint_object, pyattr_set_viewpoint_object),
+	KX_PYATTRIBUTE_BOOL_RW("autoUpdate", KX_TextureProbe, m_autoUpdate),
+	KX_PYATTRIBUTE_BOOL_RW("enabled", KX_TextureProbe, m_enabled),
+	KX_PYATTRIBUTE_INT_RW("ignoreLayers", 0, (1 << 20) - 1, true, KX_TextureProbe, m_ignoreLayers),
+	KX_PYATTRIBUTE_RW_FUNCTION("clipStart", KX_TextureProbe, pyattr_get_clip_start, pyattr_set_clip_start),
+	KX_PYATTRIBUTE_RW_FUNCTION("clipEnd", KX_TextureProbe, pyattr_get_clip_end, pyattr_set_clip_end),
+	KX_PYATTRIBUTE_FLOAT_RW("lodDistanceFactor", 0.0f, FLT_MAX, KX_TextureProbe, m_lodDistanceFactor),
+	KX_PYATTRIBUTE_NULL // Sentinel
+};
+
+KX_PYMETHODDEF_DOC_NOARGS(KX_TextureProbe, update, "update(): Set the cube map to be updated next frame.\n")
+{
+	m_forceUpdate = true;
+	Py_RETURN_NONE;
+}
+
+PyObject *KX_TextureProbe::pyattr_get_viewpoint_object(PyObjectPlus *self_v, const KX_PYATTRIBUTE_DEF *attrdef)
+{
+	KX_TextureProbe *self = static_cast<KX_TextureProbe *>(self_v);
+	KX_GameObject *gameobj = self->GetViewpointObject();
+	if (gameobj) {
+		return gameobj->GetProxy();
+	}
+	Py_RETURN_NONE;
+}
+
+int KX_TextureProbe::pyattr_set_viewpoint_object(PyObjectPlus *self_v, const KX_PYATTRIBUTE_DEF *attrdef, PyObject *value)
+{
+	KX_TextureProbe *self = static_cast<KX_TextureProbe *>(self_v);
+	KX_GameObject *gameobj = NULL;
+
+	SCA_LogicManager *logicmgr = KX_GetActiveScene()->GetLogicManager();
+
+	if (!ConvertPythonToGameObject(logicmgr, value, &gameobj, true, "cubeMap.object = value: KX_TextureProbe"))
```

Comment:
KX_TextureProbe is the base class for cubeMap/planar->reflections/refractions? The message "cubeMap.object" is only related to cubeMaps.